### PR TITLE
User/bena/install script tweaks

### PR DESCRIPTION
**Pull request by benjanderson**
use common discovery service config from shared install.config
intelligently import required modules
better defaults and error handling
prompt that it may take a while on dacpac deploy

### CODE REVIEW

File: Fabric.Terminology.API/Scripts/Install-Terminology.ps1 [hunk 1]
```diff
@@ -34,39 +34,39 @@
     [switch] $Quiet
 )
 
-[string[]] $requiredFiles = @($InstallFile, $Dacpac, $PublishProfile, "$PSScriptRoot\Install.config", "$PSScriptRoot\Register.ps1", "$PSScriptRoot\Terminology-Install-Utilities.psm1", "$PSScriptRoot\registration.config")
+# Fail installation on first error
+$ErrorActionPreference = "Stop"
```
Comment (benjanderson) [inline Fabric.Terminology.API/Scripts/Install-Terminology.ps1]: This is a built in variable in powershell

File: Fabric.Terminology.API/Scripts/Install-Terminology.ps1 [hunk 2]
```diff
@@ -34,39 +34,39 @@
...
+
+[string[]] $requiredFiles = @($InstallFile, $Dacpac, $PublishProfile, "$PSScriptRoot\Install.config", "$PSScriptRoot\Terminology-Install-Utilities.psm1", "$PSScriptRoot\registration.config")
 For ($i = 0; $i -lt $requiredFiles.Length; $i++) {
     if (!(Test-Path -Path $requiredFiles[$i])) {
         Throw "$($requiredFiles[$i]) does not exist and is required for install."
     }
 }
 
 # Import Dos Install Utilities
-$dosInstallUtilities = Get-Childitem -Path ./**/DosInstallUtilities.psm1 -Recurse
```
Comment (benjanderson) [inline Fabric.Terminology.API/Scripts/Install-Terminology.ps1]: I no longer assume DosInstallUtilties is bundled

File: Fabric.Terminology.API/Scripts/Install-Terminology.ps1 (hunk 1 above)
Comment (nickrflynn) [inline Fabric.Terminology.API/Scripts/Install-Terminology.ps1]: I don't see where this is used

Comment (nickrflynn) [reply Fabric.Terminology.API/Scripts/Install-Terminology.ps1]: Did you mean to pass it in on lines 49 and 52?

File: Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1 [hunk 3]
```diff
@@ -265,10 +275,22 @@ function Get-TerminologyConfig {
     $appNameConfig = Get-ConfigValue -Prompt "service name" -DefaultFromParam $AppName -DefaultFromInstallConfig $installSettings.appName -Quiet:$Quiet
 
     # Terminology Service Endpoint
-    $terminologyEndpointConfig = Get-ConfigValue -Prompt "$appNameConfig endpoint" -AdditionalPromptInfo "(https://SERVER/TerminologyService)" -DefaultFromParam $AppEnpoint -DefaultFromInstallConfig $installSettings.appEndpoint -Quiet:$Quiet
+    if ([string]::IsNullOrWhiteSpace($installSettings.appEndpoint)) {
+        $terminologyEndpointParam = "https://$(Get-FullyQualifiedMachineName)/TerminologyService"
```
Comment (nickrflynn) [inline Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1]: I like it!

File: Fabric.Terminology.API/Scripts/Install-Terminology.ps1 (hunk 1 above)
Comment (nickrflynn) [reply Fabric.Terminology.API/Scripts/Install-Terminology.ps1]: Copy that

**Pull request merged**